src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/RenameDriver.cs

```diff
@@ -67,10 +70,46 @@ public string ApplyRename(string expressionText)
         public bool Find(string expressionText)
         {
             var invariantExpression = _engine.GetInvariantExpression(expressionText, _baseParameters, _culture);
-
             return invariantExpression != ApplyRename(invariantExpression);
         }
 
+        private INameResolver RenameResolverHelper(INameResolver resolver, DPath pathToRename, DName updatedName)
+        {
+            if (resolver is not ReadOnlySymbolTable rost)
+            {
+                throw new Exception("Unexpected resolver");
+            }
```

Comment:
Can we push this up to and have signature just take in a ReadOnlySymbolTable?

Reply:
or actually ... do we need the cast? It looks like we go resolver --> ReadOnlySymbolTable --> resolver?